releasing/plugin-commands-deploy/test/shared-lockfile.test.ts

```diff
@@ -960,3 +961,76 @@ test('deploy with a shared lockfile that has peer dependencies suffix in workspa
     expect(readPackageJson(`deploy/node_modules/.pnpm/${name}/node_modules/project-2`)).toStrictEqual(preparedManifests['project-2'])
   }
 })
+
+test('deploy with a shared lockfile should keep files created by lifecycle scripts', async () => {
+  const preparedManifests: Record<string, ProjectManifest> = {
+    root: {
+      name: 'root',
+      version: '0.0.0',
+      private: true,
+      pnpm: {
+        neverBuiltDependencies: [],
```

Comment:
The workaround for this bug:

```suggestion
        onlyBuiltDependencies: ['@pnpm.e2e/install-script-example'],
```

However, this would reveal another bug: `pnpm deploy` fails to copy `neverBuiltDependencies` to the deploy lockfile.

Reply:
https://github.com/pnpm/pnpm/pull/8991 should fix the bug of not inheriting the right `pnpm` object. Although, the script still fails to execute.

Reply:
I see, thanks for looking into it!  
I will debug all 3 test cases of this PR again after #8991 landed 👍

Reply:
https://github.com/pnpm/pnpm/pull/8990#issuecomment-2614513135